tests/unit/misc/test_cmdhistory.py

```diff
@@ -0,0 +1,196 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2015 Alexander Cogneau (acogneau) <alexander.cogneau@gmail.com>:
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Tests for misc.History."""
+
+import pytest
+
+from qutebrowser.misc.cmdhistory import (History, HistoryEmptyError,
+                                         HistoryEndReachedError)
```

Comment:
I'd prefer it if you did `from qutebrowser.misc import cmdhistory` and used `cmdhistory.History` etc. Here it doesn't matter much, but in qutebrowser I always (with a few exceptions) import modules, and then use the functions/classes with the module names.